subsys/shell/shell_uart.c

```diff
@@ -8,6 +8,7 @@
 #include <drivers/uart.h>
 #include <init.h>
 #include <logging/log.h>
+#include <net/buf.h>
```

Comment:
buf.h has some `static inline` functions. I think it would be good to also add a `#ifdef` around this include.

Reply:
Out of 72 places found with `#include <net/buf.h>` there is none that are inside `#ifdef`. Why do you want to do it now? What is the issue with `static inline` apart from slightly more time spent in compilation?

Reply:
I am not in insisting, we can keep it as it is.

Reply:
Okay, thanks! I prefer to leave it as is then.

Reply:
> 
> Out of 72 places found with `#include <net/buf.h>` there is none that is inside `#ifdef`. Why do you want to do it now? What is the issue with `static inline` apart from slightly more time spent in compilation?

It adds up to time the CI spends on compilations and so on. Maybe we will fix these 72 other places later.